src/pages/Account/Tabs/ModulesTab/Contract.tsx

```diff
@@ -450,11 +456,70 @@ function ReadContractForm({
 
   const onSubmit: SubmitHandler<ContractFormType> = async (data) => {
     logEvent("read_button_clicked", fn.name);
-    const viewRequest: Types.ViewRequest = {
-      function: `${module.address}::${module.name}::${fn.name}`,
-      type_arguments: data.typeArgs,
-      arguments: data.args,
-    };
+    let viewRequest: Types.ViewRequest;
+    try {
+      viewRequest = {
+        function: `${module.address}::${module.name}::${fn.name}`,
+        type_arguments: data.typeArgs,
+        arguments: data.args.map((arg, i) => {
+          if (fn.params[i].includes("vector")) {
+            // encode the vector
+            const rawVector = arg.trim().slice(1, -1).split(",");
+            const regex = /vector<([^>]+)>/;
+            const match = fn.params[i].match(regex);
+            if (match) {
+              let bcsVector: any[] = [];
+              // encode the element in vector
+              bcsVector = rawVector.map((v) => {
+                switch (match[1]) {
+                  case "address": {
+                    return TxnBuilderTypes.AccountAddress.fromHex(v.trim());
+                  }
+                  case "u8": {
+                    return new TxnBuilderTypes.TransactionArgumentU8(
+                      parseInt(v.trim()),
+                    );
+                  }
+                  case "u32": {
+                    return new TxnBuilderTypes.TransactionArgumentU32(
+                      parseInt(v.trim()),
+                    );
+                  }
+                  case "u64": {
+                    return new TxnBuilderTypes.TransactionArgumentU64(
+                      BigInt(v.trim()),
+                    );
+                  }
+                  case "u128": {
+                    return new TxnBuilderTypes.TransactionArgumentU128(
+                      BigInt(v.trim()),
+                    );
+                  }
+                  case "u256": {
+                    return new TxnBuilderTypes.TransactionArgumentU256(
+                      BigInt(v.trim()),
+                    );
+                  }
+                  case "bool": {
+                    return new TxnBuilderTypes.TransactionArgumentBool(
+                      v.trim() === "true",
+                    );
+                  }
+                }
+              });
+
+              const serializer = new BCS.Serializer();
+              BCS.serializeVector(bcsVector, serializer);
+              return (HexString.fromUint8Array(serializer.getBytes()) as any)
```

Comment:
cuz the first byte is the vector length. if we get rid of the vector length, result for view is expected

Reply:
haven't figured out a better solution, let's use serializer.getBytes().subarray(1) to get rid of the bytes length prefix